src/dataflow/src/source/kafka.rs

```diff
@@ -74,6 +75,8 @@ pub struct KafkaSourceReader {
     _metadata_thread_handle: UnparkOnDropHandle<()>,
     /// A handle to the partition specific metrics
     partition_metrics: KafkaPartitionMetrics,
+    /// Whether or not to unpack and allocate headers and pass them through in the `SourceMessage`
+    include_headers: bool,
```

Comment:
This gets handled specially because of perf concerns, yeah? Like if we didn't care about perf, we'd always plumb the headers through, and let someone else drop them if `INCLUDE HEADERS` wasn't specified, yeah?

Reply:
yes, we could do this, but this is ALSO simpler to implement, because the row it produces exactly matches the above sql type, so no other layer needs to inspect that and drop the headers!